### PR TITLE
Morgue dead body placer guarantees 1 human to dissect if non-human cadaver config is enabled. Morgues spawn with 1 additional body (except on Birdboat). 

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

- Morgue guarantees 1 human body to dissect even if `morgue_cadaver_disable_nonhumans` config flag is set. 

- All maps bar birdboat will now spawn with one additional morgue cadaver. 

- Did some minor code cleanup around the dead body placer, removes an `in world` loop, etc.

## Why It's Good For The Game

- Morgue guarantees 1 human body to dissect even if `morgue_cadaver_disable_nonhumans` config flag is set. 
    - This is mostly a downstream server issue but if your server enables this config and has additional species enabled, the odds of you getting a human to dissect tends to be very low.
    - Why is this a problem? Well, a human is necessary to dissect to get medical's tech.
    - Why not get genetics to get you a hu-monkey? This is an option, but if A. there's no geneticists or B. they are refusing to cooperate then you tend to be SOL unless you want to wait for a greytide to come in after drinking themselves to death. Given we have a role now dedicated to performing dissections, having no job to do for the first twenty or so minutes due to a lack of a human body is kind of sad. 
    - If this is an intended facet, I will revert this change and leave it to the code improvements / bodycount uptick. 

- All maps bar birdboat will now spawn with one additional morgue cadaver. 
   - This was actually intended on some maps but has been stealthily removed in some cases? Icebox and Delta used to have two dead body spawners to place 4 cadavers. So I decided to bring this back.
   - For the most part, this just gives higher population maps more bodies to mess around with. Higher pop means more people means more people need bodies, either for antagging, cooking, body replacements, or coron-ing. 
   - Also like, sometimes messing around with dead bodies are fun, and it's nice to not have to worry that you're running out of them for actual medical use. 
   - I can also make this scale on roundstart pop if we really care. But that seems overkill. Especially as these maps had their body counts higher for a while and were fine. 

## Changelog

:cl: Melbert
balance: If your server has non-human morgue cadavers enabled, you will be guaranteed one human cadaver no matter what. 
balance: All maps (with the exception of Birdboat) now have an additional morgue cadaver roundstart. 
/:cl:

### CODE REVIEW

**Comment by carlarctg** [comment]
I'd call this QoL, not balance... it just makes things more likely, and the game less frustrating.

That said it'd be nice if there was SOME way to get nonhuman corpses as coroner.

**Comment by Mothblocks** [comment]
the entire config for this was misguided...the moment it was added the expectations of the non human dissection being "not trivial, but not too hard either" was disrupted

**Comment by EricZilla** [comment]
There is also making a monkey via gardening and mutadone (Which I have had to do lmao), and once when truly desperate me and the roboticist took turns killing and dissecting each other. (Bonfire Chaplain moment)

File: code/modules/mapping/mapping_helpers.dm [hunk 1]
```diff
@@ -850,23 +857,25 @@ INITIALIZE_IMMEDIATE(/obj/effect/mapping_helpers/no_lava)
...
-			if(admin_spawned)
-				var/obj/structure/bodycontainer/morgue/early_morgue_tray = i
-				if(early_morgue_tray.connected.loc != early_morgue_tray)
-					continue
-			trays += i
-	if(!trays.len)
+	var/area/morgue_area = get_area(src)
+	var/list/obj/structure/bodycontainer/morgue/trays = list()
+	for(var/turf/area_turf as anything in morgue_area.get_contained_turfs())
+		var/obj/structure/bodycontainer/morgue/morgue_tray = locate() in area_turf
```
Comment (JohnFulpWillard) [inline code/modules/mapping/mapping_helpers.dm:863]: do u want to make it not spawn in the secure morgue trays for me 🛩️

Comment (MrMelbert) [reply code/modules/mapping/mapping_helpers.dm:863]: sure

**Comment by TheVekter** [comment]
I - shockingly - don't like this change.

I think that it being predicated on Genetics not cooperating is kind of silly. I've never had a single shift where Genetics wasn't willing to hand over a single humanized monkey for my purposes so I could dissect it, given that a lot of Genetics' stuff is locked behind it just like the rest of Science.

Could this be locked behind population limits maybe? I could see the value of this on a low-pop round with no Geneticists, but otherwise it feels like removing cooperation between departments.

**Comment by Maurukas** [comment]
Not sure how I feel about this. This has never been a problem for me.

Medbay has roundstart availability of monkeys in virology, and the ability to make mutadone with about 5 clicks. That's an immediate supply of both human and non-human bodies.

**Comment by MrMelbert** [comment]
I understand it's relatively *easy* to get a human body from virology or genetics - but in worst case it ends up with the Coroner sitting on their ass for 20 minutes waiting for access (or the relative access) to stop their own setup to help you out, all the while med / sci tech is stifled. 

Ultimately I just don't think it should be in the Coroner's job description that "if you are not RNG given a human corpse you must harangue genetics or virology until they decide to leave and help you". It's "inter-department cooperation" but I don't think in a good way.

**Pull request merged**